presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonFunctions.java

```diff
@@ -88,6 +88,29 @@ public static JsonPath castCharToJsonPath(@LiteralParameter("x") Long charLength
         return new JsonPath(padSpaces(pattern, charLength.intValue()).toStringUtf8());
     }
 
+    @SqlNullable
+    @ScalarFunction("is_json_scalar")
+    @LiteralParameters("x")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static Boolean varcharIsJsonScalar(@SqlType("varchar(x)") Slice json)
+    {
+        return isJsonScalar(jsonParse(json));
+    }
+
+    @SqlNullable
```

Comment:
It looks like the only reason we return null is because some parsing error happens, which we should throw `INVALID_CAST_ARGUMENT` instead. Thus we don't need this annotation?

Reply:
If the input is a `null` (not a json null), then I guess we need to return null?

Reply:
IIRC, since you didn't mark that parameter as `SqlNullable`, the null convention for that parameter is `RETURNS_NULL_ON_NULL`, and the engine will return null without invoking the function.